python/ray/train/_internal/backend_executor.py

```diff
@@ -345,6 +346,7 @@ def start_training(
         self,
         train_func: Callable[[], T],
         datasets: Dict[str, Dataset],
+        metadata: Dict[str, Any],
```

Comment:
nit: Can we only accept `metadata` as an argument to `TrainingIterator`. Then directly pass it into `backend_executor.start_training`

no need to keep as an attr in backend executor